src/GenQPUtils.h

```diff
@@ -75,6 +75,43 @@ inline void fillQC(const std::vector<Task*>& tasks, int nrVars,
 	}
 }
 
+/**
+	* Reduce \f$ Q \f$ matrix and the \f$ c \f$ vector based on the dependencies list
+	*/
+inline void reduceQC(const Eigen::MatrixXd & QFull, const Eigen::VectorXd & CFull,
+										 Eigen::MatrixXd & Q, Eigen::VectorXd & C,
+										 const std::vector<int> & fullToReduced,
+										 const std::vector<int> & reducedToFull,
+										 const std::vector<std::tuple<int, int, double>> & dependencies)
+{
+	/* Start by moving the non-reduced variables to their new location */
+	for(size_t i = 0; i < reducedToFull.size(); ++i)
+	{
+		for(size_t j = 0; j < reducedToFull.size(); ++j)
+		{
+			Q(i,j) = QFull(reducedToFull[i], reducedToFull[j]);
+		}
+		C(i) = CFull(reducedToFull[i]);
+	}
+	/* Apply the reduction */
+	for(const auto & d : dependencies)
+	{
+		const int & primaryFullI = std::get<0>(d);
+		const int & primaryReducedI = fullToReduced[primaryFullI];
+		const int & replicaFullI = std::get<1>(d);
+		const double & alpha = std::get<2>(d);
+		/* Apply reduction to C */
+		C(primaryReducedI) += alpha*CFull(replicaFullI);
+		/* Add cross-terms to Q */
+		for(size_t i = 0; i < reducedToFull.size(); ++i)
+		{
+			Q(i, primaryReducedI) += alpha*QFull(reducedToFull[i], replicaFullI);
+			Q(primaryReducedI, i) += alpha*QFull(replicaFullI, reducedToFull[i]);
```

Comment:
I will look into it. At the very least there is indeed no need to compute `alpha*QFull(reducedToFull[i], replicaFullI)` twice